### PR TITLE
docs: update SQL in the design document

### PR DESCRIPTION
**Pull request by knz**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9651)

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -1230,9 +1230,120 @@ split to a duplicate range comprising the new configuration.
 see the protobufs in [roachpb/](https://github.com/cockroachdb/cockroach/blob/master/roachpb),
 in particular [roachpb/api.proto](https://github.com/cockroachdb/cockroach/blob/master/roachpb/api.proto) and the comments within.
 
-# Structured Data API
-
-A preliminary design can be found in the [Go source documentation](https://godoc.org/github.com/cockroachdb/cockroach/sql).
+# SQL
+
+Each node in a cluster can accept SQL client connections.  It supports
```
Comment (petermattis) [inline docs/design.md]: `s/It/CockroachDB/g`

**Review by spencerkimball — APPROVED** [review]

File: docs/design.md [hunk 2]
```diff
@@ -1230,9 +1230,120 @@ split to a duplicate range comprising the new configuration.
...
+
+CockroachDB also attempts to emulate the flavor of SQL supported by
+PostgreSQL, although it also diverges in significant ways:
+
+- CockroachDB exclusively implements MVCC-based consistency for
+  transactions, and thus only supports SQL's isolation levels SNAPSHOT
+  and SERIALIZABLE.  The other traditional SQL isolation levels are
+  not supported and there is no plan to implement them in the future.
+
+- CockroachDB implements its own SQL type system (cf. `typing.md`)
```
Comment (petermattis) [inline docs/design.md]: Can you provide a full link to `typing.md`?

File: docs/design.md [hunk 3]
```diff
@@ -1195,6 +1195,123 @@ it discovers differences, it reconfigures ranges in the same way
...
+
+## Language support
+
+CockroachDB also attempts to emulate the flavor of SQL supported by
+PostgreSQL, although it also diverges in significant ways:
+
+- CockroachDB exclusively implements MVCC-based consistency for
+  transactions, and thus only supports SQL's isolation levels SNAPSHOT
+  and SERIALIZABLE.  The other traditional SQL isolation levels are
+  not supported and there is no plan to implement them in the future.
```
Comment (tschottdorf) [inline docs/design.md]: They are supported but internally always map to either SNAPSHOT or SERIALIZABLE.

File: docs/design.md [hunk 4]
```diff
@@ -1195,6 +1195,123 @@ it discovers differences, it reconfigures ranges in the same way
...
+
+- CockroachDB implements its own SQL type system (cf. `typing.md`)
```
Comment (tschottdorf) [inline docs/design.md]: Make this a link.

File: docs/design.md (hunk 1 above)
Comment (knz) [inline docs/design.md]: done.

File: docs/design.md (hunk 2 above)
Comment (knz) [inline docs/design.md]: done.

File: docs/design.md (hunk 4 above)
Comment (knz) [inline docs/design.md]: done

File: docs/design.md (hunk 3 above)
Comment (knz) [inline docs/design.md]: done

**Pull request merged**